javascript/app-model-exposure/src/components/VegaSpecificationCurve.tsx

```diff
@@ -45,19 +45,46 @@ export const VegaSpecificationCurve: React.FC<{
 	// TODO: these two charts should be combinable into a single vega spec
 	// this will also greatly simplify the hover coordination
 	const [selected, setSelected] = useState<Maybe<string>>()
+	const [latestList, setLatestList] = useState<Specification[]>(data)
+
+	const onChangeSelectedItem = useCallback(
+		(id?: string) => {
+			setSelected(id)
+			onSpecificationSelect(data.find(d => d.id === id))
+		},
+		[data, setSelected, onSpecificationSelect],
+	)
+
 	const handleDatumClick = useCallback(
 		(item: Maybe<Specification>) => {
 			if (item && item.id === selected) {
-				setSelected(undefined)
-				onSpecificationSelect(undefined)
+				onChangeSelectedItem(undefined)
 			} else {
-				setSelected(item?.id)
-				onSpecificationSelect(data.find(d => d.id === item?.id))
+				onChangeSelectedItem(item?.id)
 			}
 		},
-		[data, selected, setSelected, onSpecificationSelect],
+		[selected, onSpecificationSelect],
 	)
 
+	const updateIds = useCallback(
```

Comment:
Why separate this into two hooks rather than put it al in the useEffect?

Reply:
I thought I was going to have undesired calls but it works, thanks!